app/org/sagebionetworks/bridge/dynamodb/DynamoUploadSchemaDao.java

```diff
@@ -82,6 +86,101 @@ public void setStudyIdIndex(DynamoIndexHelper studyIdIndex) {
+        int oldRev = getCurrentSchemaRevision(studyId, schemaId);
+
+        // Create schema.
+        DynamoUploadSchema schema = new DynamoUploadSchema();
+        schema.setFieldDefinitions(fieldDefList);
+        schema.setName(survey.getName());
+        schema.setRevision(oldRev);
+        schema.setSchemaId(schemaId);
+        schema.setSchemaType(UploadSchemaType.IOS_SURVEY);
+        schema.setStudyId(studyId);
+        UploadSchema createdSchema = createOrUpdateUploadSchema(studyId, schema);
+
+        return createdSchema;
+    }
+
+    /**
+     * Private helper function that converts a survey question constraints object into an upload schema field type.
+     * This is used to help convert surveys into upload schemas.
+     */
+    private static UploadFieldType getFieldTypeFromConstraints(Constraints constraints) {
+        if (constraints == null) {
+            // Could be anything. When in doubt, default to inline_json_blob.
+            return UploadFieldType.INLINE_JSON_BLOB;
+        }
+
+        if (constraints instanceof MultiValueConstraints
+                && ((MultiValueConstraints) constraints).getAllowMultiple()) {
+            if (constraints.getDataType() == DataType.STRING) {
+                // Multiple choice string answers frequently become longer than the 100-char limit. This will have
+                // to be a JSON attachment.
+                return UploadFieldType.ATTACHMENT_JSON_BLOB;
+            } else {
+                // Multiple choice non-string answers (frequently ints) tend to be very short, so this can fit in
+                // an inline_json_blob.
+                return UploadFieldType.INLINE_JSON_BLOB;
+            }
+        } else {
+            switch (constraints.getDataType()) {
+                case DURATION:
+                    // Upload schemas don't have this concept. Treat it as a string. (Durations are short, so this
+                    // is fine).
+                    return UploadFieldType.STRING;
+                case STRING:
+                    if (constraints instanceof StringConstraints) {
+                        Integer maxLength = ((StringConstraints) constraints).getMaxLength();
+                        if (maxLength == null || maxLength > MAX_STRING_LENGTH) {
+                            // No max length, or max length is longer than what we can fit in Synapse. This has to
+                            // be an attachment.
+                            return UploadFieldType.ATTACHMENT_BLOB;
+                        } else {
+                            // Short string. Can be a string.
+                            return UploadFieldType.STRING;
+                        }
+                    } else {
+                        // Constraints aren't StringConstraints, so we can't determine max length (if any). To be
+                        // safe, make this an attachment.
+                        return UploadFieldType.ATTACHMENT_BLOB;
```

Comment:
I believe this is very likely to be short if it's not multi-value, it will be a single boolean, int, decimal, or time/duration/datetime string. I think all would fit in under 100 characters.